packages/commons/src/context/context.ts

```diff
@@ -6,21 +6,28 @@ import { z } from "zod";
 import { AuthData, defaultAuthData } from "../auth/authData.js";
 import { readHeaders } from "../auth/headers.js";
 
-export type AppContext = z.infer<typeof ctx>;
+export type AppContext = {
+  authData: AuthData;
+  messageData?: {
```

Comment:
is `messageData` only for consumers?
if yes, can we define a different type to include the field and use the new type only in consumers?

Reply:
It is not an easy change, we should change how we handle the `AsyncLocalStorage`. Maybe it's better to make this change in another PR

Reply:
having introduced this change here it would make sense to include the refactor, but I see your point.
Which is, more or less, the effort in your opinion?

Reply:
Right now I think that we have to create a `apiContext.ts` and a `consumerContext.ts` and change the import in the various packages. 
It's not a big change but we have to touch a few files

Reply:
yes, but they could also belong to the same union type/interface, so we do not need to duplicate utils.
May I ask you to open a non-release-critical task about this?